src/http-requests.ts

```diff
@@ -111,12 +125,20 @@ class HttpRequests {
     }
 
     try {
-      const response: any = await fetch(constructURL.toString(), {
+      const fetchFn = this.httpClient ? this.httpClient : fetch
+      const _request = fetchFn(constructURL.toString(), {
```

Comment:
`const client: MeiliSearch = new MeiliSearch({
      host: 'http://localhost:3000/api/v0/meilisearch/proxy',
      httpClient: async (url, opts) => {
        const response = await $http!.getClient().request({
          url,
          data: opts?.body,
          headers: opts?.headers,
          method: (opts?.method?.toLocaleUpperCase() as Method) ?? 'GET'
        })

        return response.data
      }
    }))`
    
    in case the user provided an httpClient such as the example above

Reply:
I will add tests and examples I just wanted to make sure this is something u r open to merge first